fontbe/src/features.rs

```diff
@@ -113,39 +125,82 @@ impl<'a> FeaVariationInfo<'a> {
     }
 }
 
+/// When generating mark features it's handy to know what is a base and what is a mark
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum AnchorType {
+    Base,
+    Mark,
+}
+
+impl AnchorType {
+    fn group_name(&self, glyph_name: GlyphName) -> GlyphName {
+        match self {
+            AnchorType::Base => glyph_name,
+            AnchorType::Mark => glyph_name.as_str()[1..].into(),
+        }
+    }
+}
+
+trait AnchorInfo {
+    fn anchor_type(&self) -> AnchorType;
+}
+
+impl AnchorInfo for GlyphName {
+    fn anchor_type(&self) -> AnchorType {
+        // _ prefix means mark. This convention appears to come from FontLab and is now everywhere.
+        if self.as_str().starts_with('_') {
+            AnchorType::Mark
+        } else {
+            AnchorType::Base
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+struct MarkGroup<'a> {
+    //glyph_name: GlyphName,
+    bases: Vec<(GlyphName, &'a Anchor)>,
+    marks: Vec<(GlyphName, &'a Anchor)>,
+}
+
 struct FeatureWriter<'a> {
     kerning: &'a Kerning,
     glyph_map: &'a GlyphOrder,
     static_metadata: &'a StaticMetadata,
+    raw_anchors: &'a Vec<(FeWorkId, Arc<GlyphAnchors>)>,
 }
 
 impl<'a> FeatureWriter<'a> {
     fn new(
         static_metadata: &'a StaticMetadata,
         kerning: &'a Kerning,
         glyph_map: &'a GlyphOrder,
+        raw_anchors: &'a Vec<(FeWorkId, Arc<GlyphAnchors>)>,
     ) -> Self {
         FeatureWriter {
             kerning,
             static_metadata,
             glyph_map,
+            raw_anchors,
         }
     }
 
+    fn glyph_id(&self, glyph_name: &GlyphName) -> Option<GlyphId> {
+        self.glyph_map
+            .glyph_id(glyph_name)
+            .map(|val| Some(GlyphId::new(val as u16)))
+            .unwrap_or_default()
```

Comment:
There's an extra option or something in here, I think it should just be,
```suggestion
        self.glyph_map
            .glyph_id(glyph_name)
            .map(|val| GlyphId::new(val as u16))
```

Reply:
Good catch, I think I modified it a couple times and left it weird